src/lib/viewers/doc/DocPreloader.js

```diff
@@ -129,22 +129,18 @@ class DocPreloader extends EventEmitter {
             return;
         }
 
-        // Set image and overlay dimensions
-        setDimensions(this.imageEl, scaledWidth, scaledHeight);
-        setDimensions(this.overlayEl, scaledWidth, scaledHeight);
+        // Set initial placeholder dimensions
+        setDimensions(this.placeholderEl, scaledWidth, scaledHeight);
 
         // Add and scale correct number of placeholder elements
         for (let i = 0; i < numPages - 1; i += 1) {
             const placeholderEl = document.createElement('div');
-            placeholderEl.className = CLASS_BOX_PREVIEW_PRELOAD_CONTENT;
-            placeholderEl.innerHTML = SPINNER_HTML;
+            placeholderEl.className = CLASS_BOX_PREVIEW_PRELOAD_PLACEHOLDER;
```

Comment:
Have we considered using document fragments to append all at once instead of each iteration?

Reply:
I agree that would be an improvement, but we haven't seen any performance issues with this experience and it seems outside of the scope of this change.